test/functional/apps/discover/_doc_navigation.js

```diff
@@ -19,39 +19,28 @@
 
 import expect from '@kbn/expect';
 
-const TEST_COLUMN_NAMES = ['@message'];
-const TEST_FILTER_COLUMN_NAMES = [
-  ['extension', 'jpg'],
-  ['geo.src', 'IN'],
-];
-
 export default function({ getService, getPageObjects }) {
   const docTable = getService('docTable');
   const testSubjects = getService('testSubjects');
   const PageObjects = getPageObjects(['common', 'discover', 'timePicker']);
   const esArchiver = getService('esArchiver');
 
-  // FLAKY: https://github.com/elastic/kibana/issues/62281
-  describe.skip('doc link in discover', function contextSize() {
+  describe('doc link in discover', function contextSize() {
     before(async function() {
       await esArchiver.loadIfNeeded('logstash_functional');
       await PageObjects.common.navigateToApp('discover');
       await PageObjects.timePicker.setDefaultAbsoluteRange();
       await PageObjects.discover.waitForDocTableLoadingComplete();
-      for (const columnName of TEST_COLUMN_NAMES) {
-        await PageObjects.discover.clickFieldListItemAdd(columnName);
-      }
-      for (const [columnName, value] of TEST_FILTER_COLUMN_NAMES) {
-        await PageObjects.discover.clickFieldListItem(columnName);
-        await PageObjects.discover.clickFieldListPlusFilter(columnName, value);
-      }
     });
 
     it('should open the doc view of the selected document', async function() {
       // navigate to the doc view
       await docTable.clickRowToggle({ rowIndex: 0 });
-      await (await docTable.getRowActions({ rowIndex: 0 }))[1].click();
```

Comment:
Here is the point where some failures happened, but the screens I've got were not usefull, since they were scrolled down. In a future failure there should be a useful screen, also I thought that maybe a bit of sleep could prevent a possible race condition between opening the document detail and clicking the row action button.

Reply:
> maybe a bit of sleep could prevent a possible race condition between opening the document detail and clicking the row action button

Sounds like what you want is a `retry.waitFor(() => checkIfRowOpenedSomehow())`. Putting an arbitrary sleep in here doesn't really **prevent** anything since the timing on CI is wildly unpredictable. Only way to add safeguards is with retries that are based on specific checks.

Reply:
Actually, even better:

```
// click the open action
await retry.try(async () => {
  const rowActions = await docTable.getRowActions({ rowIndex: 0 });
  if (!rowActions.length) {
    throw new Error('row actions empty, trying again')
  }
  
  await rowActions[1].click()
})
```

Reply:
agreed and adapted the code for this case, recent flaky run also just finished successfully (https://kibana-ci.elastic.co/job/kibana+flaky-test-suite-runner/465).
I'd love to have a kind of alert when this test would fail again